py3status/py3.py

```diff
@@ -464,3 +474,16 @@ def stop_sound(self):
         if self._audio:
             self._audio.kill()
             self._audio = None
+
+    @staticmethod
+    def get_json_data(url):
+        """
+        Deserialize JSON from `url` into a Python object.
+
+        Returns None on error.
+        """
+        try:
+            data = json.loads(urlopen(url).read().decode())
+        except URLError:
```

Comment:
You are going to get many more exceptions thrown here.  json will definitely be throwing ones.

I'd do this in a much more broken down way so we can actually categorize the errors.  There are lots of different ways you can go about this.

We need to try to get this to be as robust as possible.

There is also the issue about how we handle errors.  The simple solution you have here (returning None) may well be the simplest but it doesn't give modules much to go on but maybe that is all they need.  Other options I can think of include returning a richer object like requests does so the module can get the return code etc.  I haven't thought about this in too much detail and have only looked at a couple of the modules.